src/models.ts

```diff
@@ -79,16 +108,16 @@ export const createJSONRPCErrorResponse = (
   message: string,
   data?: any
 ): JSONRPCErrorResponse => {
-  const error: JSONRPCError = { code, message };
-
-  if (data) {
-    error.data = data;
-  }
+  const error: JSONRPCErrorException = new JSONRPCErrorException(
+    code,
+    message,
+    data
+  );
```

Comment:
This looks like an avoidable overhead. Assuming you are concerned about the code duplication, how about something this instead?

```typescript
// no need to export
const createJSONRPCError = (code, message, data) => {
  const error: JSONRPCError = { code, message };
  if (data != null) { // please add != so that it supports values like 0 or an empty string
    error.data = data;
  }
  return error;
};

export class JSONRPCErrorException {
  // ...
  toObject() {
    return createJSONRPCError(this.code, this.message, this.data);
  }
  // ...
}

export const createJSONRPCErrorResponse = (/* ... */) => {
  return {
    jsonrpc: JSONRPC,
    id,
    error: createJSONRPCError(code, message, data),
  };
};
```

Reply:
I have made the change you have suggested but for my own understanding is it really that much more efficient to avoid instantiating a class or do you think it is just a bit cleaner this way?

Reply:
Both. To me, it's more about necessity than efficiency, but yes it will be additional memory allocation and `Error` class comes with a stack trace too.